packages/dd-trace/src/config.js

```diff
@@ -819,6 +723,89 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+    if (options.dogstatsd) {
+      this._setValue(opts, 'dogstatsd.port', String(options.dogstatsd.port))
+      this._setValue(opts, 'dogstatsd.hostname', options.dogstatsd.hostname)
+    }
+    this._setBoolean(opts, 'runtimeMetrics', options.runtimeMetrics)
+    this._setBoolean(opts, 'experimental.runtimeId', options.experimental && options.experimental.runtimeId)
+    this._setValue(opts, 'experimental.exporter', options.experimental && options.experimental.exporter)
+    this._setBoolean(opts, 'experimental.enableGetRumData',
+      options.experimental && options.experimental.enableGetRumData)
+    const ingestion = options.ingestion || {}
+    this._setValue(opts, 'sampler.rateLimit', coalesce(options.rateLimit, ingestion.rateLimit))
+    this._setBoolean(opts, 'reportHostname', options.reportHostname)
+    this._setBoolean(opts, 'profiling.enabled', options.profiling)
+    if (options.spanAttributeSchema) {
+      this._setValue(opts, 'spanAttributeSchema', validateNamingVersion(options.spanAttributeSchema))
+    }
+    this._setBoolean(opts, 'spanRemoveIntegrationFromService', options.spanRemoveIntegrationFromService)
+    this._setValue(opts, 'peerServiceMapping', options.peerServiceMapping)
+    this._setValue(opts, 'lookup', options.lookup)
+    this._setBoolean(opts, 'startupLogs', options.startupLogs)
+    this._setBoolean(opts, 'telemetry.logCollection', this.iastOptions &&
+    (this.iastOptions === true || this.iastOptions.enabled === true))
+    this._setValue(opts, 'protocolVersion', options.protocolVersion)
+    this._setBoolean(opts, 'appsec.enabled', this.appsecOpt.enabled)
+    this._setValue(opts, 'appsec.rules', this.appsecOpt.rules)
+    if (this.appsecOpt.rules) this._setBoolean(opts, 'appsec.customRulesProvided', !!this.appsecOpt.rules)
+    this._setValue(opts, 'appsec.rateLimit', maybeInt(this.appsecOpt.rateLimit))
+    this._setValue(opts, 'appsec.wafTimeout', maybeInt(this.appsecOpt.wafTimeout))
+    this._setValue(opts, 'appsec.obfuscatorKeyRegex', this.appsecOpt.obfuscatorKeyRegex)
+    this._setValue(opts, 'appsec.obfuscatorValueRegex', this.appsecOpt.obfuscatorValueRegex)
+    this._setValue(opts, 'appsec.blockedTemplateHtml', maybeFile(this.appsecOpt.blockedTemplateHtml))
+    this._setValue(opts, 'appsec.blockedTemplateJson', maybeFile(this.appsecOpt.blockedTemplateJson))
+    if (options.remoteConfig) {
+      this._setValue(opts, 'remoteConfig.pollInterval', maybeFloat(options.remoteConfig.pollInterval))
+    }
+    this._setBoolean(opts, 'iast.enabled',
+      this.iastOptions && (this.iastOptions === true || this.iastOptions.enabled === true))
+    const iastRequestSampling = maybeInt(this.iastOptions?.requestSampling)
+    if (iastRequestSampling > -1 && iastRequestSampling < 101) {
+      this._setValue(opts, 'iast.requestSampling', iastRequestSampling)
+    }
+    this._setValue(opts, 'iast.maxConcurrentRequests',
+      maybeInt(this.iastOptions && this.iastOptions.maxConcurrentRequests))
```

Comment:
optional chaining with the `this.iastOptions` properties?
```suggestion
      maybeInt(this.iastOptions?.maxConcurrentRequests))
```